WordPress/src/main/java/org/wordpress/android/ui/reader/services/discover/ReaderDiscoverLogic.kt

```diff
@@ -86,7 +86,14 @@ class ReaderDiscoverLogic(
                     present a static content to the user. We need to include it only for the first page (when
                     page_handle is empty). The server will take care of including the "refresh" parameter within the
                     page_handle so the user will stay on the same shard while they’re paging through the cards. */
-                    params["refresh"] = appPrefsWrapper.getReaderCardsRefreshCounter().toString()
+                    val refreshCounter = appPrefsWrapper.getReaderCardsRefreshCounter().toString()
```

Comment:
I feel like the card is not related to the server request at all and the ReaderDiscoverLogic is a class which responsibility is to sync the local db with the server db. Could we perhaps set the flag [here](https://github.com/wordpress-mobile/WordPress-Android/blob/26bc872cdd7f8ca277ab9ef7209ad508c1bbe4a2/WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderDiscoverViewModel.kt#L166)? Wdyt?

Reply:
Ohh, the getDiscoverCardsUseCase is invoked multiple times if you loadMore items. I can see the issue. Perhaps, we could clear it in viewMode.onCleared?